### PR TITLE
CLOUD-3955 - EAP XP 3.0 OpenShift image

### PR DESCRIPTION
**Pull request by luck3y**
https://issues.redhat.com/browse/CLOUD-3995
Signed-off-by: Ken Wills <kwills@redhat.com>

Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:

- [ ] Pull Request title is properly formatted: `[CLOUD-XYA] Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Attached commits represent units of work and are properly formatted
- [ ] You have read and agreed to the Developer Certificate of Origin (DCO) (see `CONTRIBUTING.md`)
- [ ] Every commit contains `Signed-off-by: Your Name <yourname@example.com>` - use `git commit -s`

### CODE REVIEW

**Comment by luck3y** [comment]
Note this still needs the final CR1 artifact.

File: eap-xp3/image.yaml [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+run:
+      user: 185
+      cmd:
+          - "/opt/eap/bin/openshift-launch.sh"
+osbs:
+      configuration:
+            container_file: container.yaml
+      repository:
+            name: containers/jboss-eap-xp2-openjdk11-builder
```
Comment (jmesnil) [inline eap-xp3/image.yaml]: this is not correct: it should be `containers/jboss-eap-xp3-openjdk11-builder`

File: eap-xp3/image.yaml [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+            branch: jb-eap-7.3-openshift-rhel-8
```
Comment (jmesnil) [inline eap-xp3/image.yaml]: correct branch is: `jb-eap-xp3-rhel-8`

File: eap-xp3/rel-j9-11-xp-overrides.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        pulp_repos: true
+        packages:
+          - java-11-openj9
+          - java-11-openj9-headless
+          - java-11-openj9-devel
+        signing_intent: release
+        inherit: true
+  repository:
+    name:  containers/jboss-eap-xp3-openj9-11-builder
+    branch: jb-eap-7.3-openshift-rhel-8
```
Comment (jmesnil) [inline eap-xp3/rel-j9-11-xp-overrides.yaml]: correct branch is: `jb-eap-xp3-rhel-8`

File: eap-xp3/rel-jdk11-xp-overrides.yaml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+            version: '7'
+
+packages:
+  manager: dnf
+  content_sets_file: content_sets_rhel8.yml
+
+osbs:
+      repository:
+            name:  containers/jboss-eap-xp3-openjdk11-builder
+            branch: jb-eap-7.3-openshift-rhel-8
```
Comment (jmesnil) [inline eap-xp3/rel-jdk11-xp-overrides.yaml]: correct branch is: `jb-eap-xp3-rhel-8`

File: eap-xp3/runtime-image/image.yaml [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+packages:
+      install:
+          # required by launch scripts
+          - hostname
+osbs:
+      configuration:
+            container_file: ../container.yaml
+      repository:
+            name: containers/jboss-eap-xp3-openjdk11-runtime
+            branch: jb-eap-7.3-openshift-rhel-8
```
Comment (jmesnil) [inline eap-xp3/runtime-image/image.yaml]: correct branch is: `jb-eap-xp3-rhel-8`

File: eap-xp3/runtime-image/rel-j9-11-xp-overrides.yaml [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        packages:
+          - java-11-openj9
+          - java-11-openj9-headless
+          - java-11-openj9-devel
+        signing_intent: release
+        inherit: true
+
+  repository:
+    name:  containers/jboss-eap-xp3-openj9-11-runtime
+    branch: jb-eap-7.3-openshift-rhel-8
```
Comment (jmesnil) [inline eap-xp3/runtime-image/rel-j9-11-xp-overrides.yaml]: correct branch is: `jb-eap-xp3-rhel-8`

File: eap-xp3/runtime-image/test/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,7 @@
+#Be sure to update eap-testapp with the name of the image generated with s2i.
+FROM jboss-eap-7/eap-xp2-openjdk11-runtime-openshift-rhel8
```
Comment (jmesnil) [inline eap-xp3/runtime-image/test/Dockerfile]: this should be: `jboss-eap-7/eap-xp3-openjdk11-runtime-openshift-rhel8`

**Review by jmesnil — CHANGES_REQUESTED** [review]
Thanks, there are misconfiguration for the containers and branches to change.

and wait for the final tag before we can build the images :)

**Comment by luck3y** [comment]
@jmesnil @jfdenise This should be all set to go now, I've dropped the hold label as well. Requires: https://github.com/jboss-container-images/jboss-eap-7-image/pull/228

File: eap-xp3/image.yaml [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+    - value: 8443
+modules:
+      repositories:
+          - name: cct_module
+            git:
+                  url: https://github.com/jboss-openshift/cct_module.git
+                  ref: 0.45.2
+          - name: jboss-eap-modules
+            git:
+                  url: https://github.com/luck3y/jboss-eap-modules.git
```
Comment (jmesnil) [inline eap-xp3/image.yaml]: this should point to the https://github.com/jboss-openshift org.

File: eap-xp3/image.yaml [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+                  ref: xp3
+          - name: jboss-eap-7-image
+            git:
+                  url: https://github.com/luck3y/jboss-eap-7-image.git
```
Comment (jmesnil) [inline eap-xp3/image.yaml]: idem

File: eap-xp3/rel-j9-11-xp-overrides.yaml [hunk 10]
```diff
@@ -0,0 +1,41 @@
+schema_version: 1
+
+name: "jboss-eap-7/eap-xp3-openj9-11-openshift-rhel8"
+description: "Red Hat JBoss Enterprise Application Platform XP 2.0 OpenShift container image."
```
Comment (jmesnil) [inline eap-xp3/rel-j9-11-xp-overrides.yaml]: should be"XP 3.0" (or do we have to override the description?)

**Review by jmesnil — CHANGES_REQUESTED** [review]

File: eap-xp3/image.yaml (hunk 8 above)
Comment (luck3y) [reply eap-xp3/image.yaml]: Ooops sorry pushed by accident.

**Review by jmesnil — APPROVED** [review]

**Pull request merged**